verification/src/tracer.rs

```diff
@@ -0,0 +1,39 @@
+use std::net::SocketAddr;
+
+use opentelemetry::{sdk::trace::Tracer, trace::TraceError};
+use tracing_subscriber::{filter::LevelFilter, layer::SubscriberExt, prelude::*};
+
+use crate::config::TracingConfiguration;
+
+pub fn init_logs(config: TracingConfiguration) {
+    let stdout = tracing_subscriber::fmt::layer().with_filter(
+        tracing_subscriber::EnvFilter::builder()
+            .with_default_directive(LevelFilter::INFO.into())
+            .from_env_lossy(),
+    );
+    let registry = tracing_subscriber::registry()
+        // output logs (tracing) to stdout with log level taken from env (default is INFO)
+        .with(stdout);
+    if config.enable_jaeger {
+        let tracer = init_jaeger_tracer(config.jaeger_agents).expect("failed to init tracer");
+        registry
+            // output traces to jaeger with default log level (default is DEBUG)
+            .with(
+                tracing_opentelemetry::layer()
+                    .with_tracer(tracer)
+                    .with_filter(LevelFilter::DEBUG),
+            )
+            .try_init()
+    } else {
+        registry.try_init()
+    }
```

Comment:
don't like this way, but Idk how to make it without else

Reply:
doesn't this work
```rust
let registry = if config.enable_jaeger {
  let tracer = ...;
  registry.with(...)
} else {
  registry
}
registry.try_init().expect(..);
```

Reply:
Of course I tried this, but problem is that `.with()` takes self and returns not Self but `SomeStructure<Self>`, and you cannot case `registry` to `registry.with(...)`

Reply:
well it seems you can only do this using dyn (if you even can), which in your case is actually worse imo
but I'll provide it jfyi
something like
```rust
let registry: Box<dyn SubscriberInitExt> = Box::new(
  if ... {
    let tracer = ...;
    registry.with(..) as dyn SubscriberInitExt
  } else {
    registry as dyn SubscriberInitExt
  }
);
```
or something like that